airflow/operators/python.py

```diff
@@ -145,6 +149,131 @@ def execute_callable(self):
         return self.python_callable(*self.op_args, **self.op_kwargs)
 
 
+class _PythonFunctionalOperator(BaseOperator):
+    """
+    Wraps a Python callable and captures args/kwargs when called for execution.
+
+    :param python_callable: A reference to an object that is callable
+    :type python_callable: python callable
+    :param op_kwargs: a dictionary of keyword arguments that will get unpacked
+        in your function
+    :type op_kwargs: dict (templated)
+    :param op_args: a list of positional arguments that will get unpacked when
+        calling your callable
+    :type op_args: list (templated)
+    :param multiple_outputs: if set, function return value will be
+        unrolled to multiple XCom values. Dict will unroll to xcom values with keys as keys.
+        Defaults to False.
+    :type multiple_outputs: bool
+    """
+
+    template_fields = ('op_args', 'op_kwargs')
+    ui_color = '#ffefeb'
+
+    # since we won't mutate the arguments, we should just do the shallow copy
+    # there are some cases we can't deepcopy the objects(e.g protobuf).
+    shallow_copy_attrs = ('python_callable',)
+
+    @apply_defaults
+    def __init__(
+        self,
+        python_callable: Callable,
+        op_args: Tuple[Any],
+        op_kwargs: Dict[str, Any],
+        multiple_outputs: bool = False,
+        **kwargs
+    ) -> None:
+        kwargs['task_id'] = self._get_unique_task_id(kwargs['task_id'], kwargs.get('dag', None))
+        super().__init__(**kwargs)
```

Comment:
```suggestion
        super().__init__(**kwargs)
        self.task_id = self._get_unique_task_id(kwargs['task_id'], kwargs.get('dag', None))
```

Reply:
No strong opinion here, I just think this is more explicit

Reply:
Nvm, seems this proposed change fails due to task_id being checked if repeated on ` super().__init__(**kwargs)`

Resolving.

Reply:
I see another issue here. Currently, if `task_id` is not provided user will get `KeyError: 'task_id'` instead of TypeError: __init__() missing 1 required positional argument: 'task_id'